libraries/AP_Scripting/applets/param-set-filter.lua

```diff
@@ -0,0 +1,150 @@
+-- Inspect parameter sets received via MAVLink, determine action based
+-- on whitelist
+
+-- How To Use
+-- 1. copy this script to the autopilot's "scripts" directory
+-- 2. within the "scripts" directory create a "modules" directory
+-- 3. copy the mavlink/mavlink_msgs_xxx files to the "scripts" directory in a "modules/mavlink" folder
+-- 4. set PSF_ENFORCE to disallow sets
+
+-- load mavlink message definitions from modules/mavlink directory
+local mavlink_msgs = require("mavlink/mavlink_msgs")
+
+-- global definitions
+local MAV_SEVERITY = {EMERGENCY=0, ALERT=1, CRITICAL=2, ERROR=3, WARNING=4, NOTICE=5, INFO=6, DEBUG=7}
+local UPDATE_INTERVAL_MS = 10           -- update at about 100hz
+
+-- get a reference to MAV_SYSID parameter so we can filter messages:
+MAV_SYSID = Parameter("MAV_SYSID")
+
+-- mavlink definitions
+local PARAM_SET_ID = 23
+local msg_map = {}
+msg_map[PARAM_SET_ID] = "PARAM_SET"
+
+local TEXT_PREFIX_STR = "param-set-filter"    -- prefix for all text messages
+
+--
+-- parameter setup
+--
+local PARAM_TABLE_KEY = 92
+local PARAM_TABLE_PREFIX = "PSF_"
+assert(param:add_table(PARAM_TABLE_KEY, PARAM_TABLE_PREFIX, 7), 'could not add param table')
+
+-- add a parameter and bind it to a variable
+function bind_add_param(name, idx, default_value)
+    assert(param:add_param(PARAM_TABLE_KEY, idx, name, default_value), string.format('could not add param %s', PARAM_TABLE_PREFIX .. name))
+    return Parameter(PARAM_TABLE_PREFIX .. name)
+end
+
+--[[
+  // @Param: PSF_ENABLE
+  // @DisplayName: Packet Set Filter enable
```

Comment:
Maybe it's Param set instead of Packet set?

Reply:
Yes, thanks.

... and this isn't really a "filter" now, as such, it's really an alternate setting mechanism.  So I'm probably going to have to rename this - I'm going to try dropping the "filter" bit.